kotlin-generator/src/main/scala/models/generator/kotlin/KotlinUtil.scala

```diff
@@ -150,7 +150,7 @@ trait KotlinUtil {
     "double" -> new ClassName("kotlin","Double"),
     "integer" -> new ClassName("kotlin", "Int"),
     "long" -> new ClassName("kotlin", "Long"),
-    "object" -> new ClassName("kotlin.collections","Map"),
+    "object" -> new ClassName("kotlin","Any"),
```

Comment:
Same as above.

Reply:
Should be Map<String,Any>?

Reply:
@sullis what about this?

Reply:
@gheine this is broken in both android/java and kotlin if it's supposed to be Map<String, Obj>, and we thought we will address the issue in a separate commit.

What's the reasoning behind having it be Map<String,Object>?
The value could be a nested json object too, and this only gives us one layer of confidence.

Java generator does: object -> Map<String, String>
Scala generator does: object -> _root_.play.api.libs.json.JsObject

Reply:
The apibuilder object type represents json objects, where you either don't care or don't know anything more specific about the contents to use a proper model type or the keys are not static. As per the json spec, an object is a dictionary/hash with string keys and values that can be any valid json value, see https://www.json.org/

The question is then, how to represent that in a given language mapping. scala/play-json uses JsObject (which is directly equivalent to Map[String,JsValue]).

For java and kotlin you could either tie yourself into a particular json parsing library, say jackson, and represent object as Map<String,JsonNode>. Or you don't, in which case the Map values need to be a super-type of all possible java representations of json values, (eg. String,Number,etc.) which has to be java.lang.Object. 

If you use the jackson ObjectMapper uder the hoods and deserialize a given json into Map<String,Object> the "Object" values will be one of "String,Number,Boolean,Array[Object],Map<String,Object>,null" which allows you to traverse nested objects and arrays via introspection.

Hope that helps. Fwiw, I've just submitted a PR to add support for "json value" types, which would map to JsValue or java.lang.Object or JsonNode etc. respectively, see https://github.com/apicollective/apibuilder/pull/715